### PR TITLE
Add Manifest.toml for documentation generation

### PR DESCRIPTION
**Pull request by mkitti**

### CODE REVIEW

**Comment by musm** [comment]
So the Manifest.toml fixes the issue since it specifies the relative directories to pull in?

File: docs/Manifest.toml [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+
+[[deps.H5Zzstd]]
+deps = ["CodecZstd", "HDF5"]
+path = "../filters/H5Zzstd"
+uuid = "f6f2d980-1ec6-471c-a70d-0270e22f1103"
+version = "0.1.0"
+
+[[deps.HDF5]]
+deps = ["Compat", "HDF5_jll", "Libdl", "Mmap", "Random", "Requires"]
+path = ".."
```
Comment (musm) [inline docs/Manifest.toml:125]: Ah I see the relative paths  are in the manifest file. Ok I think I understand how this is working.  

Will this then always just use the latest  commits for each package?

Comment (mkitti) [reply docs/Manifest.toml:125]: We are essentially "dev"ing them into the project.

**Comment by mkitti** [comment]
https://juliaio.github.io/HDF5.jl/previews/PR903/filters/

File: docs/Manifest.toml (hunk 1 above)
Comment (musm) [reply docs/Manifest.toml:125]: Ok makes sense. Yeah I guess only one package that would've happened automatically, but with the subpacakges it seems like we have to force it to 'dev' them throught a Manifest.toml file.

**Pull request merged**